### ISSUE TITLE
Thread are not working again (v4.1.3)

### ISSUE THREAD

**Issue by clavelm** [TwidereProject/Twidere-Android#1327]
See closed issues for v4.1.0 
#1272 
#1289 
Same problem again for v4.1.3

(sorry I didn't put all the info on that new issue)

**Comment by Tlaster** [comment]
Could you please post some tweet's link here that has the issue in order to resolve the issue?
Seems like it's working here.

**Comment by clavelm** [comment]
Sorry, I can't find one at the moment. It's threads where there is a lot of retweets with comments on the tweets of the thread and the rest of the thread is missing. I will update this issue as soon as I find one.

**Comment by Tlaster** [comment]
Thanks, now I know what's happening.

> but there is 3 responses from other people before seeing the 8th message of the thread.

> I only see 10 responses from other people and that's all, I can't load more than 10 responses.

I think these are the issue that related to Twidere, others are Twitter's issues I think, I can reproduce then on Twitter's official client, or am I missing something?

I think we can add an option to switch to nitter in order to fetch all replies and threads, currently, we can only load up to 10 replies in some tweets since Twidere is fetching replies from Twitter's legacy web page.

**Comment by Xefir** [comment]
Just out of curiosity, why Twidere does not fetch replies and thread from API instead of curling Twitter website ?

**Comment by Tlaster** [comment]
I don't see any API for Twitter to fetch tweet reply and thread, or am I missing something?

**Comment by Xefir** [comment]
You're right, I though that there was an API for that but seems like there's nothing

**Comment by clavelm** [comment]
I'll check using Twitter in my browser, but I think they load threads as it should, responses from the same user are priorized. I don't know how Twidere gets the tweets from Twitter, api or scraping of the web site (as NewPipe do for YouTube) but if it's the latter, it should be possible to emulate the same behavior than the Twitter website, shouldn't it?

**Comment by clavelm** [comment]
> Thanks, now I know what's happening.
> 
> > but there is 3 responses from other people before seeing the 8th message of the thread.
> 
> > I only see 10 responses from other people and that's all, I can't load more than 10 responses.
> 
> I think these are the issue that related to Twidere, others are Twitter's issues I think, I can reproduce then on Twitter's official client, or am I missing something?
> 
> I think we can add an option to switch to nitter in order to fetch all replies and threads, currently, we can only load up to 10 replies in some tweets since Twidere is fetching replies from Twitter's legacy web page.

What I can say using mobile.twitter.com in Firefox for Android.

When loading the 8th message, 1~7 are displayed before, the 0th is missing, so it's not a Twidere problem.

When loading the 7th, the 1~6 are displayed before, the 0th is still missing (while Twidere displays it) so I think Twitter ignores it because of the time discrepancy between the 0th and the 1st (8h).
The 8th message is displayed just after the 7th.

When loading the 0th, the thread is not displayed, it's consistent that this message is not part of the thread for Twitter.

When loading one of the the 1~8 messages, there is a title at the top of the Twitter pages : Discussion (French for thread).

When loading the last message of a longer thread in Twitter, all the previous messages from the thread are loaded. Same with Twidere (see French thread https://twitter.com/reesmarc/status/1270558042162114562?s=20 for example). So the missing 0th thread is not a Twidere problem. What is strange is why it's missing when displaying the 8th message, but present when loading the 7th.

So Twitter identifies and assembles all the messages from threads (messages responding to the previous one of the same user, in a specific time span for all  messages or a max time between messages), but there is no API for it, that's what you are saying?

So no flag or thread id gotten with a message to know it's part of a thread and no API request with a thread id to get all and only the thread messages.

The only way is to scrap the website or use API from other sources. They really want users to use the official site / client for the ads I think.

**Comment by Tlaster** [comment]
> So Twitter identifies and assembles all the messages from threads (messages responding to the previous one of the same user, in a specific time span for all messages or a max time between messages), but there is no API for it, that's what you are saying?

Yes, I can not find the API to fetching reply and thread. if there is please let me know.

> So no flag or thread id gotten with a message to know it's part of a thread and no API request with a thread id to get all and only the thread messages.

There is a flag called "in_reply_to_status_id" from Twitter API, but it's not a "thread id". Twidere uses that to load the thread's tweets by getting each tweet recursively.

**Comment by Rudloff** [comment]
There is a feature request for this: https://twitterdevfeedback.uservoice.com/forums/921790-twitter-developer-labs/suggestions/37629283-build-an-api-to-create-and-retrieve-tweet-threads
